app/main.go

```diff
@@ -61,8 +59,26 @@ func getIPAddress(r *http.Request) string {
 }
 
 func handlerPing(w http.ResponseWriter, r *http.Request) {
-	w.Write([]byte("pong"))
-	fmt.Println(time.Now(), getIPAddress(r), r.Method, r.RequestURI, r.UserAgent())
+	
+	requestBody, err := ioutil.ReadAll(r.Body)
+	n := rand.Intn(100)
+
+	select {
+		case <- time.After(500 * time.Millisecond):
+		if n <= 30 {
+			w.Write([]byte("pong"))
+			fmt.Println(time.Now(), getIPAddress(r), r.Method, r.RequestURI, r.UserAgent())
+		} else {
```

Comment:
In this case we don't need to use `else`. This rule applies: https://thanos.io/tip/contributing/coding-style-guide.md/#control-structure-prefer-early-returns-and-avoid-else

It will allow bit more readable code in practice if we avoid else if possible (less indentations)